### PR TITLE
new format ivf (VP8): still very simple, no frame disassembly, but it…

### PR DESCRIPTION
**Pull request by frankpolte**
… is a start. works with the ivf files from https://github.com/webmproject/vp8-test-vectors

### CODE REVIEW

File: media/ivf.ksy [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## header start
+  - id: magic1
+    contents: DKIF
+    doc: Magic Number of IVF Files
+  - id: version
+    type: u2
+    doc: This should be 0
+  - id: headerlen
+    type: u2
+    doc: Normally the header length is 32 byte
```
Comment (GreyCat) [inline media/ivf.ksy]: Could you make this `len_header` to match [style guide](https://doc.kaitai.io/ksy_style_guide.html#attr-id) recommendation?

File: media/ivf.ksy [hunk 2]
```diff
@@ -0,0 +1,67 @@
+meta:
+  id: ivf
+  file-extension: ivf
+  endian: le
+  title: VP8 raw file format
```
Comment (GreyCat) [inline media/ivf.ksy]: Please remove word "format" from the title — just "VP8 raw file" will work, as in certain contexts this "title" would be used as `$title format`.

File: media/ivf.ksy [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    doc: The (initial) height of the video, every keyframe may change the resolution
+  - id: framerate
+    type: u4
+    doc: the (framerate * timescale) e.g. for 30 fps -> 30000 
+  - id: timescale
+    type: u4
+    doc: the timescale is a divider of the seconds (VPX is integer math only) mostly 1000
+  - id: framecount
+    type: u4
+    doc: the number of frames (if not a camera stream)
```
Comment (GreyCat) [inline media/ivf.ksy]: `num_frames`?

File: media/ivf.ksy [hunk 4]
```diff
@@ -0,0 +1,67 @@
+meta:
+  id: ivf
```
Comment (GreyCat) [inline media/ivf.ksy]: Formats repo have global IDs without any namespacing. Given that "ivf" is not a very popular format and there might be other formats identified as "ivf", may I suggest to rename this (and the file) to something more verbose, like `vp8_ivf`, as this is clearly a container for VP8?

File: media/ivf.ksy [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  # eithder a chunk or list
+  blocks:
+    seq:
+      - id: entries
+        type: block
+        #repeat: eos
+        #size-eos: true
+  block:
+    seq:
+      - id: framesize
```
Comment (GreyCat) [inline media/ivf.ksy]: `len_frame`?

**Comment by frankpolte** [comment]
@GreyCat : I renamed and cleaned up the stuff. Even the file was renamed, so that we can add other codec ivf files e.g. vp9_ivf and so on. Right now it is for all codecs, but I plan for parsing vp8 specifically. Is there a way to reference the outer stuff from a more global ksy file?
Maybe a `ivf_mediacontainer.ksy` holds the main header stuff and `vp8_ivf.ksy`, `vp9_ivf.ksy` or `xyz_ivf.ksy` only parse the frame content but including  `ivf_mediacontainer.ksy`.

File: media/vp8_ivf.ksy [hunk 6]
```diff
@@ -0,0 +1,64 @@
+meta:
+  id: vp8_ivf
+  file-extension: ivf
+  endian: le
+  title: VP8 raw file
+  ks-version: 0.7
+  license: CC0-1.0
+doc-ref: https://wiki.multimedia.cx/index.php/IVF
+seq:
+## header start
```
Comment (KOLANICH) [inline media/vp8_ivf.ksy]: I usually prefer the code be self-documenting. Should you remove these comments and use subtypes instead?

File: media/vp8_ivf.ksy [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  - id: magic1
+    contents: DKIF
+    doc: Magic Number of IVF Files
+  - id: version
+    type: u2
+    doc: This should be 0
+  - id: len_header
+    type: u2
+    doc: Normally the header length is 32 byte
+  - id: codec
```
Comment (KOLANICH) [inline media/vp8_ivf.ksy]: I don't think it is meant to be hardcoded.

**Comment by GreyCat** [comment]
> Is there a way to reference the outer stuff from a more global ksy file?

Sure, we have  [imports mechanism](http://doc.kaitai.io/user_guide.html#_importing_types_from_other_files) for that.

I guess everything looks good now, so I'm merging it. Thanks again!

**Pull request merged**